src/core/namespace.c

```diff
@@ -1790,6 +1792,16 @@ static int apply_one_mount(
                 break;
         }
 
+        case MOUNT_EMPTY_DIR_WHEN_EMPTY:
+                r = dir_is_empty(mount_entry_path(m), /* ignore_hidden_or_backup= */ false);
```

Comment:
I'm not convinced. This basically means what the service gets for /var/ is racy for no good. Any future mounts cannot be propagated into namespace anymore. And upon later restart they might get completely different stuff...

Like, if `RootImage=` is used, what's mounted in /var/ _is_ stable, regardless of PrivateTmp=. Now, there's no guarantee that services on host, even if not explicitly ordered after /var/, doesn't need to access it. It's fair to assume that /var/ will be mounted later on, and they can monitor until that happens and operate on that. In fact, networkd-persistent-storage is backed by a similar idea, just we used fd passing so this is a non-issue.

Frankly, let's not be overzealous and make up problems.

Reply:
Even then we're still left with the race part. I don't see the rationale behind this...

Reply:
This check is not really useful btw. If /var/ is not going to be a mount yet still is empty, we'd mount a tmpfs there pointlessly.